x-pack/plugins/security_solution/public/management/pages/event_filters/view/components/form.tsx

```diff
@@ -81,169 +78,141 @@ const getAddedFieldsCounts = (formFields: string[]): { [k: string]: number } =>
-          isEqual(arg.exceptionItems[0]?.entries, exception?.entries)
-        ) {
-          const addedFields = arg.exceptionItems[0]?.entries.map((e) => e.field) || [''];
-          setHasDuplicateFields(computeHasDuplicateFields(getAddedFieldsCounts(addedFields)));
-          setHasFormChanged(true);
-          return;
-        }
-        setHasFormChanged(true);
-        dispatch({
-          type: 'eventFiltersChangeForm',
-          payload: {
-            ...(arg.exceptionItems[0] !== undefined
-              ? {
-                  entry: {
-                    ...arg.exceptionItems[0],
-                    name: exception?.name ?? '',
-                    description: exception?.description ?? '',
-                    comments: exception?.comments ?? [],
-                    os_types: exception?.os_types ?? [OperatingSystem.WINDOWS],
-                    tags: exception?.tags ?? [],
-                  },
-                  hasItemsError: arg.errorExists || !arg.exceptionItems[0]?.entries?.length,
-                }
-              : {
-                  hasItemsError: true,
-                }),
-          },
-        });
-      },
-      [dispatch, exception, hasFormChanged]
-    );
+    const eventFilterItem = useMemo<ArtifactFormComponentProps['item']>(() => {
+      const ef: ArtifactFormComponentProps['item'] = exception;
+      ef.entries = exception.entries.length
+        ? (exception.entries as ExceptionListItemSchema['entries'])
+        : defaultConditionEntry();
 
-    const handleOnChangeName = useCallback(
-      (e: React.ChangeEvent<HTMLInputElement>) => {
-        if (!exception) return;
-        setHasFormChanged(true);
-        const name = e.target.value.toString().trim();
-        dispatch({
-          type: 'eventFiltersChangeForm',
-          payload: {
-            entry: { ...exception, name },
-            hasNameError: !name,
-          },
-        });
-      },
-      [dispatch, exception]
-    );
+      // TODO: `id` gets added to the exception.entries item
+      // Is there a simpler way to this?
+      cleanupEntries(ef);
 
-    const handleOnDescriptionChange = useCallback(
-      (e: React.ChangeEvent<HTMLTextAreaElement>) => {
-        if (!exception) return;
-        setHasFormChanged(true);
-        const description = e.target.value.toString().trim();
-        dispatch({
-          type: 'eventFiltersChangeForm',
-          payload: {
-            entry: { ...exception, description },
-          },
-        });
-      },
-      [dispatch, exception]
-    );
+      setAreConditionsValid(!!exception.entries.length);
+      return ef;
+    }, [exception]);
 
-    const handleOnChangeComment = useCallback(
-      (value: string) => {
+    // name and handler
+    const handleOnChangeName = useCallback(
+      (event: React.ChangeEvent<HTMLInputElement>) => {
         if (!exception) return;
+        const name = event.target.value.trim();
+        setHasNameError(!name);
```

Comment:
Not sure I understand. `setHasNameError` is actually toggling the state with a `!` operator. Since the `onChange` event triggers on input change this whole callback method is executed only on change. Consequently, the state is updated every time a change occurs. Let me know if renaming this update function to `toggleHasNameError` might make it easier to understand?

Also, the setHasFormChanged should be set to true when this handler is called. No?

Reply:
Yeah, maybe the `toggleHasNameError` is a better function name. About the `setHasFormChanged` it should be set only the first time:

```Typescript
if (!hasFormChanged) setHasFormChanged(true);
```

Reply:
https://github.com/elastic/kibana/pull/130995/commits/e165d70a035e265a7ac3129a8b437b7c7ec3cd90

Reply:
https://github.com/elastic/kibana/pull/130995/commits/e165d70a035e265a7ac3129a8b437b7c7ec3cd90